src/swarm-util/Swarm/Util/CNF.hs

```diff
@@ -0,0 +1,48 @@
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Utilities for dealing with conjunctive normal form.
+module Swarm.Util.CNF (
+  minimizePCNF,
+) where
+
+import Data.List (sortBy)
+import Data.Ord (comparing)
+import Data.Set (Set)
+import Data.Set qualified as S
+import Data.Set.NonEmpty (NESet)
+import Data.Set.NonEmpty qualified as NES
+
+type PCNF a = Set (NESet a)
+
+-- | Minimize an expression in positive conjunctive normal form.  That
```

Comment:
I have a hunch that the `boolexpr` package implements something in like this.

Reply:
Not really.  The `boolexpr` package does have a data type to represent expressions in conjunctive normal form, but with *signed* variables.  Minimizing CNF expressions *in general* is NP-hard (I'm pretty sure) and the `boolexpr` package does not have any such functions.

Reply:
But wait...!!  *We* already implement something like this... in `Swarm.Util.removeSupersets`. :facepalm: